heron/api/src/java/com/twitter/heron/dsl/Streamlet.java

```diff
@@ -163,4 +163,12 @@
    * of this streamlet.
    */
   void consume(SerializableConsumer<R> consumer);
+
+  /**
+   * Applies the sink's put function to every element of the stream
+   * This function does not return anything.
+   * @param sink The Sink whose put method consumes each element
+   * of this streamlet.
+   */
+  void toSink(Sink<R> sink);
```

Comment:
Is the ‘toSink’ method general enough that needed by every streamlet?

Is this new method here for stateful processing?

Reply:
The toSink is just for abstracting out complex sinks(aka terminator bolts) that might need a setup/cleanup.Not really related to stateful processing.